enterprise/frontend/src/embedding-sdk/components/public/CollectionBrowser/CollectionBrowser.tsx

```diff
@@ -0,0 +1,65 @@
+import { useState } from "react";
+
+import { withPublicComponentWrapper } from "embedding-sdk/components/private/PublicComponentWrapper";
+import {
+  ALL_MODELS,
+  COLLECTION_PAGE_SIZE,
+} from "metabase/collections/components/CollectionContent";
+import { CollectionItemsTable } from "metabase/collections/components/CollectionContent/CollectionItemsTable";
+import CollectionBreadcrumbs from "metabase/nav/containers/CollectionBreadcrumbs/CollectionBreadcrumbs";
+import type {
+  CollectionEssentials,
+  CollectionId,
+  CollectionItem,
+  CollectionItemModel,
+} from "metabase-types/api";
+
+type CollectionBrowserProps = {
+  collectionId: CollectionId;
+  onClick?: (item: CollectionItem) => void;
+  pageSize?: number;
+  visibleCollectionTypes?: CollectionItemModel[];
+};
+
+export const CollectionBrowser = withPublicComponentWrapper(
+  ({
+    collectionId,
+    onClick,
+    pageSize = COLLECTION_PAGE_SIZE,
+    visibleCollectionTypes = ALL_MODELS,
+  }: CollectionBrowserProps) => {
+    const [currentCollectionId, setCurrentCollectionId] =
```

Comment:
We use this state as the collection being currently viewed. Then we can use the given `collectionId` as the starting point.